groovy/inductor/inductor_xpu_backend_e2e.groovy

```diff
@@ -0,0 +1,304 @@
+        '''
+    }//stage
+    stage('Install Dependency') {
+        println('================================================================')
+        println('Install Dependency')
+        println('================================================================')
+            sh '''
+            set -e
+            set +x
+            source ${HOME}/miniconda3/etc/profile.d/conda.sh 2>&1 >> /dev/null
+            conda activate ${conda_env}
+            source ${HOME}/env.sh 
+            python -m pip install numpy --upgrade --no-cache --force-reinstall
+
+            if [ ! -d inductor-tools ];then
+                git clone --depth=1 https://github.com/chuanqi129/inductor-tools.git
+            fi
+
+            if [[ -n ${torch_whl} ]] && [[ -n ${ipex_whl} ]];then
+                python -m pip uninstall -y torch || true
+                wget -q -e use_proxy=no ${torch_whl}
+                python -m pip install --force-reinstall $(basename ${torch_whl})
+                python -m pip uninstall -y intel_extension_for_pytorch || true
+                wget -q -e use_proxy=no ${ipex_whl}
+                python -m pip install --force-reinstall $(basename ${ipex_whl})
+            else
+                bash ${WORKSPACE}/inductor-tools/scripts/env_prepare.sh
+                ${torch_repo} \
+                ${torch_branch} \
+                ${torch_commit} \
+                ${ipex_repo} \
+                ${ipex_branch} \
+                ${ipex_commit} 
+                source ${HOME}/env.sh
+                python -c "import torch;import intel_extension_for_pytorch"
+                if [ ${PIPESTATUS[0]} -ne 0 ]; then
+                    echo -e "[ERROR] Public-torch or IPEX BUILD FAIL"
+                    exit 1
+                fi
+            fi
+            '''
+    }//stage
+    stage('Build Triton') {
+        println('================================================================')
+        println('Build Triton')
+        println('================================================================')
+        retry(1){
+            sh'''
+            set -e
+            set +x
+            source ${HOME}/miniconda3/etc/profile.d/conda.sh 2>&1 >> /dev/null
+            conda activate ${conda_env}
+            source ${HOME}/env.sh 
+            pip uninstall -y triton
+            sudo update-ca-certificates --fresh
+            export SSL_CERT_DIR=/etc/ssl/certs
+            pip install pybind11
+            if [[ -n ${triton_whl} ]] ;then
+                pip uninstall -y triton || true
+                wget -q -e use_proxy=no ${triton_whl}
+                python -m pip install --force-reinstall $(basename ${triton_whl})
+            else
+                git clone https://github.com/openai/triton triton
+                cd triton
+                git submodule sync
+                git submodule update --init --recursive --jobs 0
+                cd third_party/intel_xpu_backend
+                git checkout main && git pull
```

Comment:
If we need build triton from scratch, we'd better add a xpu backend branch & commit parameters in here to make the job more flexible